docs/src/model_setup/boundary_conditions.md

```diff
@@ -488,3 +488,49 @@ Oceananigans.FieldBoundaryConditions, with boundary conditions
 └── immersed: ImmersedBoundaryCondition with west=Nothing, east=Nothing, south=Nothing, north=Nothing, bottom=Value, top=Nothing
 ```
 
+A boundary condition that depends on the fields may be prescribed using the `immersed` keyword argument in [`FieldBoundaryConditions`](@ref). Let's see here how we can add linear bottom drag on `u` in the case we have
+an immersed boundary. In essence, we'd need to boundary conditions: one that would apply at the grid's bottom,
+and when that would apply on the immersed boundary.
+
+First let's create the boundary condition for the grid's bottom:
+
+```jldoctest immersed_bc
+julia> @inline linear_drag(x, y, t, u) = - 0.2 * u
+linear_drag (generic function with 1 method)
+
+julia> drag_u = FluxBoundaryCondition(linear_drag, field_dependencies=:u)
+FluxBoundaryCondition: ContinuousBoundaryFunction linear_drag at (Nothing, Nothing, Nothing)
+```
+
+Then let's create the immersed boundary condition:
+
+```jldoctest immersed_bc
+julia> @inline immersed_linear_drag(x, y, z, t, u) = - 0.2 * u
```

Comment:
I think something that may not be intuitive and that might be worth calling attention to is that the function for the bottom only depends on `(x, y, t, u)`, but the one for the immersed boundary also depends on `z`.

Reply:
Good point!

Reply:
I'd just point out that neither is "intuitive", they simply are what they are. It'd good to point out this difference though.

Reply:
resolved by [d00e735](https://github.com/CliMA/Oceananigans.jl/pull/2632/commits/d00e73581a91e0479411ee0c818ce6f201143a55)